### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Harrynoqs**
- Create a class called Solver.

- Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.

- Create a method called reverse that takes one argument, a string word, and returns word reversed.

- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:

    - When N is divisible by 3, return "fizz".
    - When N is divisible by 5, return "buzz".
    - When N is divisible by 3 and 5, return "fizzbuzz".
    - Any other case, return N as a string ).

We start with tests for all of the methods described above. Our commit history proves that we have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+RSpec.describe Solver do
+  context 'Factorial' do
+    it 'Test : Factorial of a number' do
+      result = Solver.new
+      expect(result.factorial(2)).to eq 2
+      expect(result.factorial(3)).to eq 6
+      expect(result.factorial(1)).to eq(1)
+      expect(result.factorial(-2)).to eq('No negative number')
+    end
```
Comment (mahbubul14) [inline spec/solver_spec.rb]: - [x] Kindly add a special case - when the input is 0, the result should equal 1. :+1:

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @Team,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [x] `Gemfile` & `Gemfile.lock` files are missing from your project. kindly fix it. :+1: 

- [x] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [x] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "Fixed bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/solver_spec.rb (hunk 1 above)
Comment (CakeInTech) [reply spec/solver_spec.rb]: Thanks for the reminder @mahbubul14 It's done. ✅ 😄

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @Harrynoqs, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**